lib/mongoid/config/defaults.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Config
+
+    # Encapsulates logic for loading defaults.
+    module Defaults
+
+      # Load the defaults for the feature flags in the given Mongoid version.
+      # Note that this method will load the *new* functionality introduced in
+      # the given Mongoid version.
+      #
+      # @param [ String | Float ] The version number as X.y.
+      #
+      # raises [ ArgumentError ] if an invalid version is given.
+      def load_defaults(version)
+        # Note that for 7.x, since all of the feature flag defaults have been
+        # flipped to the new functionality, all of the settings for those
+        # versions are to give old functionality. Because of this, it is
+        # possible to recurse to later version to get all of the options to
+        # turn off. Note that this won't be true when adding feature flags to
+        # 8.1, since the default will be the old functionality until the next
+        # major version is released. More likely, the recursion will have to go
+        # in the other direction (towards earlier versions).
+
+        case version.to_s
```

Comment:
This should explicitly round to 1 decimal place before the `to_s` call for good measure.

Or maybe more than one but there should be some sort of rounding to a certain precision performed.

Reply:
I followed Rails' lead on this one... The parameter can also be a string and it feels awkward to convert to a number, round, and convert back to a string.

Reply:
If that is how Rails does it it can't be wrong.